beacon_node/http_api/src/lib.rs

```diff
@@ -2148,6 +2149,58 @@ pub fn serve<T: BeaconChainTypes>(
             },
         );
 
+    // GET debug/fork_choice
+    let get_debug_fork_choice = eth_v1
+        .and(warp::path("debug"))
+        .and(warp::path("fork_choice"))
+        .and(warp::path::end())
+        .and(chain_filter.clone())
+        .and_then(|chain: Arc<BeaconChain<T>>| {
+            blocking_json_task(move || {
+                let beacon_fork_choice = chain.canonical_head.fork_choice_read_lock();
+
+                let proto_array = beacon_fork_choice.proto_array().core_proto_array();
+
+                let fork_choice_nodes = proto_array
+                    .nodes
+                    .iter()
+                    .map(|node| {
+                        let execution_status = if node.execution_status.is_execution_enabled() {
+                            Some(node.execution_status.to_string())
+                        } else {
+                            None
+                        };
+
+                        Ok(ForkChoiceNode {
+                            slot: node.slot,
+                            block_root: node.root,
+                            parent_root: node
+                                .parent
+                                .and_then(|index| proto_array.nodes.get(index))
+                                .map(|parent| parent.root),
+                            justified_epoch: node
+                                .justified_checkpoint
+                                .map(|checkpoint| checkpoint.epoch),
+                            finalized_epoch: node
+                                .finalized_checkpoint
+                                .map(|checkpoint| checkpoint.epoch),
+                            weight: node.weight,
+                            validity: execution_status,
+                            execution_block_hash: node
+                                .execution_status
+                                .block_hash()
+                                .map(|block_hash| block_hash.into_root()),
+                        })
+                    })
+                    .collect::<Result<Vec<_>, warp::Rejection>>();
```

Comment:
It seems that the function passed to `map` can't error, so we may as well collect directly into a `Vec` and remove the `Result`.

```suggestion
                    .collect::<Vec<_>>();
```